sycl-fusion/common/include/Kernel.h

```diff
@@ -61,6 +61,39 @@ enum class ParameterKind : uint32_t {
 /// Different binary formats supported as input to the JIT compiler.
 enum class BinaryFormat : uint32_t { INVALID, LLVM, SPIRV, PTX, AMDGCN };
 
+/// Unique ID for each supported architecture in the SYCL implementation.
+using DeviceArchitecture = unsigned;
```

Comment:
Cast `sycl::ext::oneapi::experimental::architecture` to `unsigned` is not correct. 
`sycl::ext::oneapi::experimental::architecture` enum is always changing, e.g.,
if it was:
```
intel_gpu_abc,
intel_gpu_def,
```
it can easily be
```
intel_gpu_abc,
intel_gpu_new_arch, // <------
intel_gpu_def,
```
so anytime new architecture can be added in the middle of enum, and this cast from release to release will produce different results, so the app won't be backward-compatible and I assume caching will not work correctly. 
And as `sycl::ext::oneapi::experimental::architecture` extension is in the experimental stage, any other parts of it can be changed anytime with API/ABI break.

Reply:
The compilation here happens at runtime, not application compile time. The cache is also not persisted between different runs of the application, so backwards compatibility is not a concern here.

Reply:
Lukas is right. This will take the values from the current run, no need to make this backwards compatible, we just needed a UID for the target architectures.

Reply:
Can you please add a comment that this value is used only in non-persistent cache and only for current application run, so must not be used for creating the value which will be used between application runs?

Reply:
I think this comment is enough. Thanks for pointing out!